### PR TITLE
Move TableConstraint to Constraints conversion

### PR DESCRIPTION
**Pull request by findepi**
Reduce datafusion-common dependency on sqlparser

### CODE REVIEW

**Review by goldmedal — APPROVED** [review]
Thanks @findepi it makes sense to me 👍 . Just one minor suggestion.

File: datafusion/sql/src/statement.rs [hunk 1]
```diff
@@ -1262,6 +1263,74 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         )))
     }
 
+    /// Convert each `TableConstraint` to corresponding `Constraint`
```
[the remark this answers is not in the document]
Comment (findepi) [reply datafusion/sql/src/statement.rs:1266]: -> https://github.com/apache/datafusion/pull/12995

**Comment by findepi** [comment]
thanks for review @goldmedal and the merge @alamb !

> I wonder how far we are from being able to remove the dependency entirely:
> 
> [datafusion/datafusion/common/Cargo.toml](https://github.com/apache/datafusion/blob/44127ec96d7f2be2200af03ad65a2af1b8340b0a/datafusion/common/Cargo.toml#L65)


That's a good question. Actually this is where i'd like to get to eventually. It should be possible to use datafusion without pulling sqlparser-rs dependency, which anyone with their own grammar should not use.


There is couple places where this is used.


- `DataFusionError::SQL` -- this can be put behind a feature cfg, or maybe we add some dyn trait to `DataFusionError` to allow for dynamic binding
- identifiers:
  - `Column::from_qualified_name_ignore_case`
  - `Column::from_qualified_name`
  - `TableReference::parse_str`

@alamb what would you suggest?

**Comment by alamb** [comment]
> @alamb what would you suggest?

>  this can be put behind a feature cfg, or maybe we add some dyn trait to DataFusionError to allow for dynamic binding

A feature cfg is probably best I suppose

> identifiers:

We could make a trait in datafusion-sql maybe

Though now that I see this list, I think maybe the extra code overhead of breaking this out won't really add much so perhaps it is not worth doing anything more at this time

**Pull request merged**